src/transformers/models/swinv2/convert_swinv2_timm_to_pytorch.py

```diff
@@ -0,0 +1,205 @@
+# coding=utf-8
+# Copyright 2022 The HuggingFace Inc. team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Convert Swinv2 checkpoints from the timm library."""
+
+import argparse
+import json
+
+import torch
+from PIL import Image
+
+import requests
+import timm
+from huggingface_hub import hf_hub_download
+from transformers import AutoFeatureExtractor, Swinv2Config, Swinv2ForImageClassification
+
+
+def get_swinv2_config(swinv2_name):
+    config = Swinv2Config()
+    name_split = swinv2_name.split("_")
+
+    model_size = name_split[1]
+    if "to" in name_split[3]:
+        img_size = int(name_split[3][-3:])
+    else:
+        img_size = int(name_split[3])
+    if "to" in name_split[2]:
+        window_size = int(name_split[2][-2:])
+    else:
+        window_size = int(name_split[2][6:])
+
+    if model_size == "tiny":
+        embed_dim = 96
+        depths = (2, 2, 6, 2)
+        num_heads = (3, 6, 12, 24)
+    elif model_size == "small":
+        embed_dim = 96
+        depths = (2, 2, 18, 2)
+        num_heads = (3, 6, 12, 24)
+    elif model_size == "base":
+        embed_dim = 128
+        depths = (2, 2, 18, 2)
+        num_heads = (4, 8, 16, 32)
+    else:
+        embed_dim = 192
+        depths = (2, 2, 18, 2)
+        num_heads = (6, 12, 24, 48)
+
+    if "to" in swinv2_name:
+        config.pretrained_window_sizes = (12, 12, 12, 6)
+
+    if ("22k" in swinv2_name) and ("to" not in swinv2_name):
+        num_classes = 21841
```

Comment:
Note that we also do provide the id2label mappings for ImageNet-21k, would be great to set them as well for these models

Reply:
Added.